app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,20 @@
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+ 
+  def perform(initiator, benchmark_type, benchmark_result_type, category)
+    commit_objects = BenchmarkRun.where( initiator: initiator, benchmark_type: benchmark_type,
+                     benchmark_result_type: benchmark_result_type).limit(1000)
```

Comment:
Can another benchmark result be submitted so fast? Because the only race condition will be while fetching the data which itself is quite fast.